src/mser.rs

```diff
@@ -0,0 +1,171 @@
+                   max_variation: f64,
+                   min_diversity: f64,
+                   max_evolution: i32,
+                   area_threshold: f64,
+                   min_margin: f64,
+                   edge_blur_size: i32
+    ) -> *mut CMSER;
+    fn cv_mser_drop(cmser: *mut CMSER);
+    fn cv_mser_detectRegions(cmser: *mut CMSER, image: *mut CMat, msers: *mut CVecOfPoints, bboxes: *mut CVecOfRect);
+}
+
+/// Maximally stable extremal region extractor.
+#[derive(Debug)]
+pub struct MSER {
+    value: *mut CMSER,
+}
+
+impl MSER {
+    /// Creates a new maximally stable extremal region extractor criteria.
+    pub fn new(delta: i32,
+               min_area: i32,
+               max_area: i32,
+               max_variation: f64,
+               min_diversity: f64,
+               max_evolution: i32,
+               area_threshold: f64,
+               min_margin: f64,
+               edge_blur_size: i32
+    ) -> Self {
+        let mser = unsafe { cv_mser_new(delta,
+                                           min_area,
+                                           max_area,
+                                           max_variation,
+                                           min_diversity,
+                                           max_evolution,
+                                           area_threshold,
+                                           min_margin,
+                                           edge_blur_size) };
+        MSER { value: mser }
+    }
+
+    /// Detect MSER regions.
+    pub fn detect_regions(&self, image: &Mat) -> (Vec<Vec<Point2i>>, Vec<Rect>) {
+        let mut msers = CVecOfPoints::default();
+        let mut bboxes = CVecOfRect::default();
+        unsafe {
+            cv_mser_detectRegions(self.value, image.inner, &mut msers, &mut bboxes);
+        }
+        let msers = msers.rustify();
+        let boxes =  bboxes.rustify();
+        (msers, boxes)
+    }
+}
+
+impl Drop for MSER {
+    fn drop(&mut self) {
+        unsafe {
+            cv_mser_drop(self.value);
+        }
+    }
+}
+
+/// Builder that provides defaults for MSER
+#[derive(Debug, Copy, Clone)]
+pub struct MSERBuilder {
+    delta: Option<i32>,
+    min_area: Option<i32>,
+    max_area: Option<i32>,
+    max_variation: Option<f64>,
+    min_diversity: Option<f64>,
+    max_evolution: Option<i32>,
+    area_threshold: Option<f64>,
+    min_margin: Option<f64>,
+    edge_blur_size: Option<i32>
+}
+
+impl MSERBuilder {
+    /// Creates new MSER builder
+    pub fn new() -> Self {
+        MSERBuilder {
+            delta: None,
+            min_area: None,
+            max_area: None,
+            max_variation: None,
+            min_diversity: None,
+            max_evolution: None,
+            area_threshold: None,
+            min_margin: None,
+            edge_blur_size: None,
+        }
+    }
+
+    /// Replace current delta with specified value
+    pub fn delta(mut self, value: i32) -> Self{
```

Comment:
Space in `Self {`.

The whole project should have been formatted with `cargo fmt`... But that should happen in a different PR.

Reply:
@nebgnahz if you want to use `cargo fmt` I also suggest to create `rustfmt.toml` to persist non-default rules (if any) in the repo itself.

Reply:
Sure!